planner/core/src/test/java/eu/seaclouds/platform/planner/core/DamGeneratorTest.java

```diff
@@ -124,5 +125,112 @@ public void testGroupsAsTopologyChild() throws Exception {
         assertTrue(topologyGroups.containsKey("sla_gen_info"));
     }
 
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testNuroDAM() throws Exception {
+        String adp = new Scanner(new File(Resources.getResource("nuro_adp.yml").toURI())).useDelimiter("\\Z").next();
+
+        when(fakeAgreementManager.generateAgreeemntId(((Map<String, Object>) anyObject())))
+                .thenReturn(FAKE_AGREEMENT_ID);
+
+        DamGenerator damGenerator = new DamGenerator(MONITOR_URL, MONITOR_PORT, SLA_ENDPOINT, INFLUXDB_URL, INFLUXDB_PORT);
+        damGenerator.setAgreementManager(fakeAgreementManager);
+        dam = damGenerator.generateDam(adp);
+        template = (Map<String, Object>) yamlParser.load(dam);
+
+        String expectedDAMString = new Scanner(new File(Resources.getResource("nuro_dam.yml").toURI())).useDelimiter("\\Z").next();
+        Map<String, Object> expectedDAM = (Map<String, Object>) yamlParser.load(expectedDAMString);
+
+        assertNotNull(template);
+
+        Map<String, Object> generatedTopologyTemplate = (Map<String, Object>) template.get(DamGenerator.TOPOLOGY_TEMPLATE);
+        Map<String, Object> expectedTopologyTemplate = (Map<String, Object>) expectedDAM.get(DamGenerator.TOPOLOGY_TEMPLATE);
+
+        assertEquals(generatedTopologyTemplate.get("www"), expectedTopologyTemplate.get("www"));
+        assertEquals(generatedTopologyTemplate.get("db"), expectedTopologyTemplate.get("db"));
+        assertEquals(generatedTopologyTemplate.get("Amazon_EC2_m1_small_eu_central_1"), expectedTopologyTemplate.get("Amazon_EC2_m1_small_eu_central_1"));
+        assertEquals(generatedTopologyTemplate.get("Amazon_EC2_m4_10xlarge_eu_west_1"), expectedTopologyTemplate.get("Amazon_EC2_m4_10xlarge_eu_west_1"));
+
+        Map<String, Object> generatedGroups = (Map<String, Object>) generatedTopologyTemplate.get(DamGenerator.GROUPS);
+        Map<String, Object> expectedGroups = (Map<String, Object>) expectedTopologyTemplate.get(DamGenerator.GROUPS);
+
+        assertEquals(generatedGroups.get("operation_www"), expectedGroups.get("operation_www"));
```

Comment:
Each property should be tested. I think `tosca.nodes.Compute` NodeTypes properties could avoided in the test but the rest of the NodeTemplates should be reviewed.
Policies and their properties should also be tested.